feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java

```diff
@@ -142,9 +142,9 @@ public void start(Stage stage) {
       // the features visible in the viewpoint extent every time the map is redrawn
       featureLayer.addDoneLoadingListener(() -> {
         if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
-          mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
+          mapView.drawStatusProperty().addListener((property, oldValue, newValue) -> {
             // if the draw status is in progress, show the progress indicator and set the count label text
-            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS) {
+            if (newValue == DrawStatus.IN_PROGRESS) {
```

Comment:
I can't add comments in the right lines, but I think the listener change is fine except we can remove the progress indicator and have a binding there.

So add this above the feature layer listener:
```
// display progress indicator if drawing is in progress
progressIndicator.visibleProperty().bind(mapView.drawStatusProperty().isEqualTo(DrawStatus.IN_PROGRESS));
```

The comment on line 146 can also be changed and should also be moved to inside the if statement:

```
if (newValue == DrawStatus.IN_PROGRESS) {
   // if the draw status is in progress set the count label text
```